src/storage/v3/shard_rsm.cpp

```diff
@@ -101,31 +118,239 @@ std::optional<std::map<PropertyId, Value>> CollectAllPropertiesFromAccessor(
+
+  for (const auto &name : property_names) {
+    properties.push_back(dba.NameToProperty(name));
+  }
+  return properties;
+}
+
+std::vector<memgraph::storage::v3::LabelId> NamesToLabels(const std::vector<std::string> &label_names,
+                                                          DbAccessor &dba) {
+  std::vector<memgraph::storage::v3::LabelId> labels;
+  labels.reserve(label_names.size());
+  for (const auto &name : label_names) {
+    labels.push_back(dba.NameToLabel(name));
+  }
+  return labels;
+}
+
+template <class TExpression>
+auto Eval(TExpression *expr, EvaluationContext &ctx, AstStorage &storage,
+          memgraph::storage::v3::ExpressionEvaluator &eval, DbAccessor &dba) {
+  ctx.properties = NamesToProperties(storage.properties_, dba);
+  ctx.labels = NamesToLabels(storage.labels_, dba);
+  auto value = expr->Accept(eval);
+  return value;
+}
+
+std::vector<PropertyId> GetPropertiesFromAcessor(
+    const std::map<memgraph::storage::v3::PropertyId, memgraph::storage::v3::PropertyValue> &properties) {
+  std::vector<PropertyId> ret_properties;
+  ret_properties.reserve(properties.size());
+
+  std::transform(properties.begin(), properties.end(), std::back_inserter(ret_properties),
+                 [](const auto &prop) { return prop.first; });
+
+  return ret_properties;
+}
+
+std::any ParseExpression(const std::string &expr, memgraph::expr::AstStorage &storage) {
+  memgraph::frontend::opencypher::Parser<memgraph::frontend::opencypher::ParserOpTag::EXPRESSION> parser(expr);
+  ParsingContext pc;
+  CypherMainVisitor visitor(pc, &storage);
+
+  auto *ast = parser.tree();
+  return visitor.visit(ast);
+}
+
+TypedValue ComputeExpression(DbAccessor &dba, const std::optional<memgraph::storage::v3::VertexAccessor> &v_acc,
+                             const std::optional<memgraph::storage::v3::EdgeAccessor> &e_acc,
+                             const std::string &expression, std::string_view node_name, std::string_view edge_name) {
+  AstStorage storage;
+  Frame frame{1 + 1};  // 1 for the node_identifier, 1 for the edge_identifier
+  SymbolTable symbol_table;
+  EvaluationContext ctx;
+
+  ExpressionEvaluator eval{&frame, symbol_table, ctx, &dba, View::OLD};
+  auto expr = ParseExpression(expression, storage);
+
+  auto node_identifier = Identifier(std::string(node_name), false);
+  bool is_node_identifier_present = false;
+  auto edge_identifier = Identifier(std::string(edge_name), false);
+  bool is_edge_identifier_present = false;
+
+  std::vector<Identifier *> identifiers;
+
+  // TODO: use a visitor instead of string search
+  if (v_acc && expression.find(node_name) != std::string::npos) {
+    is_node_identifier_present = true;
+    identifiers.push_back(&node_identifier);
+  }
+  if (e_acc && expression.find(edge_name) != std::string::npos) {
+    is_edge_identifier_present = true;
+    identifiers.push_back(&edge_identifier);
+  }
+
+  expr::SymbolGenerator symbol_generator(&symbol_table, identifiers);
```

Comment:
```suggestion
  expr::SymbolGenerator symbol_generator(&symbol_table, identifiers);
```

The symbol generator receives the "predefined identifiers", which means they don't have to be matched in the cypher query, it is meant to used only with triggers, so I think they are not needed here, furthermore they are faulty, because those identifiers can only be used once in the expression, so `n.age + n.height` might not work.